src/frontend/README.md

```diff
@@ -1,131 +1,31 @@
-| `/robots.txt`     | *      | Search engine response (disallow) |
-| `/_healthz`       | *      | Health check (ok)                 |
-
-These requests are instrumented in the main function as part of the router's definition.
-
-```go
-    // Add OpenTelemetry instrumentation to incoming HTTP requests controlled by the gorilla/mux Router.
-    r.Use(otelmux.Middleware("server"))
-```
-
-### gRPC instrumentation
-
-This service will issue several outgoing gRPC calls, which have instrumentation
-hooks added in the `mustConnGRPC` function.
+Currently, the easiest way to run the frontend for local development is to execute
 
-```go
-func mustConnGRPC(ctx context.Context, conn **grpc.ClientConn, addr string) {
-    // Add OpenTelemetry instrumentation to outgoing gRPC requests
-    var err error
-    ctx, cancel := context.WithTimeout(ctx, time.Second*3)
-    defer cancel()
-    *conn, err = grpc.DialContext(ctx, addr,
-        grpc.WithTransportCredentials(insecure.NewCredentials()),
-        grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
-        grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor()),
-    )
-    if err != nil {
-        panic(errors.Wrapf(err, "grpc: failed to connect %s", addr))
-    }
-}
+```shell
+  docker compose run --service-ports -e NODE_ENV=development
+  --volume $(pwd)/src/frontend:/app --volume $(pwd)/pb:/app/pb frontend sh
 ```
 
-### Service specific instrumentation attributes
-
-All requests incoming to the frontend service will receive the following attributes:
-
-- `app.session.id`
-- `app.request.id`
-- `app.currency`
-- `app.user.id` (when the user is present)
+from the root folder.
 
-These attributes are added in the `instrumentHandler` function (defined in the
-middleware.go file) which wraps all HTTP routes specified within the
-gorilla/mux router.
-Additional attributes are added within each handler's function as appropriate
-(ie: `app.cart.size`, `app.cart.total.price`).
+It will start all of the required backend services
+and within the container simply run `npm run dev`.
+After that the app should be available at <localhost:8080>
 
-Adding attributes to existing auto-instrumented spans can be accomplished by
-getting the current span from context, then adding attributes to it.
+## Building the Docker image
 
-```go
-    span := trace.SpanFromContext(r.Context())
-    span.SetAttributes(
-        attribute.Int(instr.AppPrefix+"cart.size", cartSize(cart)),
-        attribute.Int(instr.AppPrefix+"cart.items.count", len(items)),
-        attribute.Float64(instr.AppPrefix+"cart.shipping.cost", shippingCostFloat),
-        attribute.Float64(instr.AppPrefix+"cart.total.price", totalPriceFloat),
-    )
-```
-
-When an error is encountered, the current span's status code and error message
-are set.
-
-```go
-    // set span status on error
-    span := trace.SpanFromContext(r.Context())
-    span.SetStatus(codes.Error, errMsg)
-```
+Before committing your changes to the repo,
```

Comment:
I feel this should be moved to the `CONTRIBUTING.md` doc? Is the intention here to make the `README.md` a mixture of contents which target different audiences?

Reply:
I removed it, for the time being, we can add a section about it in the `CONTRIBUTING.md` file latter on

Reply:
Thank you @xoscar.